ulauncher/__init__.py

```diff
@@ -8,10 +8,10 @@
 def _exec_get_(*args):
     try:
         return check_output(list(args)).decode('utf-8').rstrip()
-    except FileNotFoundError:
+    except (FileNotFoundError, CalledProcessError):
         return ""
 
 
 __data_directory__ = f'{Path(__file__).resolve().parent.parent}/data'
 __is_dev__ = True
-__version__ = _exec_get_("git", "describe", "--tags") or _exec_get_("./setup.py", "--version") or "DEV"
+__version__ = _exec_get_("git", "describe", "--tags") or "DEV"
```

Comment:
`git describe --tags` assumes the version is a git repo. So if it's not it will be "DEV"
This means for example if someone downloads the tarball source and build it will load "DEV" as the version

Reply:
yeah the important bit though is that `./setup.py --version` looks at `ulauncher/__init__.__version__` which tries to run ` `./setup.py --version`  which looks at `ulauncher/__init__.__version__`.... and so on until your memory is exhausted ;) 

In my case `git describe --tags` didn't work because I use work trees, and the dev container fails because it can only see the worktree and not the full tree.  But no matter the reason, if `git describe` fails, we can't then call `./setup.py --version`, because the only time that *can* work is:

* `git describe` actually works, but then we wouldn't have this problem to start with (e.g `./setup.py --version` doesn't get called)
* __version__ was already overwritten with a static value, in which case we didn't need to call `./setup.py --version` in the first place

Reply:
I see. I figured out how to load the version from setup.cfg with configparser instead, so I can avoid those circular calls. I'll try to fix that later and ask you to review it so I won't sneak in another bug for you :facepalm: